### ISSUE TITLE
Window on Windows 10 is not centered.

### ISSUE THREAD

**Issue by pegasusearl** [f3d-app/f3d#1590]
Launching F3D in Windows 10, the window is not centered, but instead in the top left corner.

![image](https://github.com/user-attachments/assets/625d6481-e839-4171-bcd6-e188aa3dd048)

Putting them in top left corner making the title bar inaccessible due being covered by taskbar.
But even if the title bar is accessible I think it's better to put them at the center of the screen of the currently focused display.

**Comment by mwestphal** [comment]
The position of the window is not set by F3D but by the OS itself. If you open F3D multiple time you will see that the windows are positioned "nicely" so that all their title bar is visible, this is Windows doing its thing.

The taskbar stuff looks like Windows not taking taskbar position when computating new Window position.

You can force the position of the F3D window using the `--position` option, and you can set this option in your config file for a permanent change.

**Comment by mwestphal** [comment]
> For other apps on Windows, the behavior is also different. For example:

I don't doubt it :)

F3D however tries to not touch anything but keep defaults from the system. That being said, on MacOS we do position the window because it we do not, it is really wrongly positioned, maybe an arguement can be made that the window should be positioned somewhere else by default on Windows.

Centering the window is not trivial as it requires access to the screen size.

> No this didn't happen. Every single f3d window is opened in top left corner (under taskbar) no matter how many instances I opened.

Indeed, I just checked and it doesnt do that. I wonder why as I remembers this behavior clearly from some years ago, Ill check.

> Perhaps this is more like feature request than a bug,

So your feature request would be that, F3D should position itself by default in the center of the screen, is that correct ?

**Comment by mwestphal** [comment]
related: https://github.com/f3d-app/f3d/issues/410

**Comment by mwestphal** [comment]
Interestingly, some Windows app do have the cascading effect (eg, Win11 terminal app), but others dont (cmake-gui), and some even do not support to be opened multiple times (eg: clock)

I do thing we could improve this behavior to be more Windows-like. It is quite barebone now. I'll add it to the Windows related improvements: https://github.com/f3d-app/f3d/issues/21

**Comment by pegasusearl** [comment]
# Primary Request

> So your feature request would be that, F3D should position itself by default in the center of the screen, is that correct ?

Yes, but more specifically:
- f3d's window, by default, should be positioned in the **center** of the screen of currently focused **display.**

I believe this is the ideal/best behavior. It is important that it's only centered in the current focused display in case user is using multi monitor setup, and they are browsing 3D assets on their secondary monitor.

Something like this: (it's two monitors stacked vertically)

![2024-08-26 21_57_56--fs8](https://github.com/user-attachments/assets/a3b0cacb-d1e7-4bdb-b156-84576935d31b)

## Possible Issue:
I'm not sure how to get the screen size, but if the screen size obtained is span of multiple monitors, then if there are two monitors the window will be created in between two monitors. I remembered some apps do this in the past, I don't remember which one. This should **NOT** be what happened.

What I meant is something like this: (I apologize for the wallpaper)

![span](https://github.com/user-attachments/assets/fcb78135-f61a-4918-a360-903ccc2d9327)

---------------------------------------------------

# Alternative Request!

In case previous request is not possible, perhaps `remember_last_window_position_and_size` would be good too. Personally I think this is less ideal compared to previous solution because you have to drag the window to the desired position/size each time you open the app with different setup/intent but some people might find it better this way.

This behavior is similar to Nomacs image viewer.

## Possible Issue:
If user have multiple monitor, and they drag f3d to secondary display, close it, and then f3d saves it's position and size, then unplug the second monitor, the next time they open f3d, the window will be gone. It will be opened outside of the screen and you can't see them.

**Comment by mwestphal** [comment]
> f3d's window, by default, should be positioned in the center of the screen of currently focused display.

Indeed, that would be nice, but position should be controllable too if needed.

> remember_last_window_position_and_size

https://github.com/f3d-app/f3d/issues/856

**Comment by Ursus63** [comment]
Hello. Is option to remember window size and position still in progress or I'm doing something wrong?

**Comment by mwestphal** [comment]
> Hello. Is option to remember window size and position still in progress or I'm doing something wrong?

Still in progress, you can track it here:  https://github.com/f3d-app/f3d/issues/856

You can hard code the position if you want though.
